packages/server/graphql/mutations/helpers/pushEstimateToGitHub.ts

```diff
@@ -0,0 +1,226 @@
+    const {repository} = issueRes
+    if (!repository) return new Error('Repository not found on GitHub')
+
+    const {issue} = repository
+    if (!issue) return new Error('Issue not found on GitHub')
+
+    const {id: issueId} = issue
+    const {name: meetingName, phases} = meeting
+    const estimatePhase = getPhase(phases, 'ESTIMATE')
+    const {stages} = estimatePhase
+    const stageIdx = stages.findIndex((stage) => stage.id === stageId)
+    const discussionURL = makeAppURL(appOrigin, `meet/${meetingId}/estimate/${stageIdx + 1}`)
+    const body = makeScoreGitHubComment(
+      dimensionName,
+      value || '<None>',
+      meetingName,
+      discussionURL
+    )
+
+    const [, commentError] = await githubRequest<AddCommentMutation, AddCommentMutationVariables>(
+      addComment,
+      {
+        input: {
+          body,
+          subjectId: issueId
+        }
+      }
+    )
+    if (commentError) return commentError
+    return undefined
+  }
+
+  const [repoLabelsRes, repoLabelsError] = await githubRequest<
+    GetRepoLabelsQuery,
+    GetRepoLabelsQueryVariables
+  >(getRepoLabels, {
+    name: repoName,
+    owner: repoOwner,
+    issueNumber,
+    first: 100
+  })
+  if (repoLabelsError) return repoLabelsError
+
+  const {repository} = repoLabelsRes
+  if (!repository) return new Error('Repository not found on GitHub')
+
+  const {id: repositoryId, labels, issue} = repository
+  if (!labels) return new Error('Labels for repository not found')
+  if (!issue) return new Error(`Issue #${issueNumber} not found`)
+
+  const {id: issueId, labels: issueLabels} = issue
+  const {nodes} = labels
+  if (!nodes) return new Error('Label nodes not found')
+  if (!issueLabels) return new Error('Issue labels not found')
+
+  const {nodes: nullableIssueLabelNodes} = issueLabels
+  if (!nullableIssueLabelNodes) return new Error('Issue label nodes not found')
+
+  const issueLabelNodes = nullableIssueLabelNodes.filter(isNotNull)
+  const labelNodes = nodes.filter(isNotNull)
+  const matchingLabel = labelNodes.find((node) => node.name === githubLabelName)
+  const DESCRIPTION_IDENTIFIER = 'Generated by Parabol'
+  let labelToAddId = matchingLabel?.id ?? ''
+  if (!matchingLabel) {
+    let color = PALETTE.GRAPE_500.slice(1)
+    if (meeting) {
+      const {templateRefId} = meeting
+      const templateRef = await dataLoader.get('templateRefs').load(templateRefId)
+      const {dimensions} = templateRef
+      const dimensionRef = dimensions.find((dimension) => dimension.name === dimensionName)
+      if (!dimensionRef) return new Error('Dimension not found')
+      const {scaleRefId} = dimensionRef
+      const scaleRef = await dataLoader.get('templateScaleRefs').load(scaleRefId)
+      const {values} = scaleRef
+      const matchingValue = values.find((scaleValue) => scaleValue.label === value)
+      if (matchingValue) {
+        // remove # prefix
+        color = matchingValue.color.slice(1)
```

Comment:
The label being the same colour as the card is absolute 🔥 . After removing the score from the label, I noticed that the colour doesn't update with the selected card. That seems like the right move as the label is no longer tied to the card and the user may have selected a specific colour for that label name

Reply:
yeah, i figure we provide the reasonable defaults but after that they're free to plot their own demise